src/api/VitessceConfig.js

```diff
@@ -0,0 +1,401 @@
+          ? Object.keys(this.config.coordinationSpace[cType])
+          : []
+      );
+      const scope = new VitessceConfigCoordinationScope(cType, getNextScope(prevScopes));
+      if (!this.config.coordinationSpace[scope.cType]) {
+        this.config.coordinationSpace[scope.cType] = {};
+      }
+      this.config.coordinationSpace[scope.cType][scope.cScope] = scope;
+      result.push(scope);
+    });
+    return result;
+  }
+
+  /**
+   * Set the layout of views.
+   * @param {VitessceConfigView|VitessceConfigViewHConcat|VitessceConfigViewVConcat} viewConcat A
+   * view or a concatenation of views.
+   * @returns {VitessceConfig} This, to allow chaining.
+   */
+  layout(viewConcat) {
+    function layoutAux(obj, xMin, xMax, yMin, yMax) {
+      const w = xMax - xMin;
+      const h = yMax - yMin;
+      if (obj instanceof VitessceConfigView) {
+        obj.setXYWH(xMin, yMin, w, h);
+      } else if (obj instanceof VitessceConfigViewHConcat) {
+        const { views } = obj;
+        const numViews = views.length;
+        views.forEach((view, i) => {
+          layoutAux(view, xMin + (w / numViews) * i, xMin + (w / numViews) * (i + 1), yMin, yMax);
+        });
+      } else if (obj instanceof VitessceConfigViewVConcat) {
+        const { views } = obj;
+        const numViews = views.length;
+        views.forEach((view, i) => {
+          layoutAux(view, xMin, xMax, yMin + (h / numViews) * i, yMin + (h / numViews) * (i + 1));
+        });
+      }
+    }
+
+    layoutAux(viewConcat, 0, 12, 0, 12);
+
+    return this;
+  }
+
+  /**
+   * Convert this instance to a JSON object that can be passed to the Vitessce component.
+   * @returns {object} The view config as a JSON object.
+   */
+  toJSON() {
+    return {
+      ...this.config,
+      datasets: this.config.datasets.map(d => d.toJSON()),
+      coordinationSpace: fromEntries(
+        Object.entries(this.config.coordinationSpace).map(([cType, cScopes]) => ([
+          cType,
+          fromEntries(
+            Object.entries(cScopes).map(([cScopeName, cScope]) => ([
+              cScopeName,
+              cScope.cValue,
+            ])),
+          ),
+        ])),
+      ),
+      layout: this.config.layout.map(c => c.toJSON()),
+    };
+  }
+
+  /**
+   * Create a VitessceConfig instance from an existing view config, to enable
+   * manipulation with the JavaScript API.
+   * @param {object} config An existing Vitessce view config as a JSON object.
+   * @returns {VitessceConfig} A new config instance, with values set to match
+   * the config parameter.
+   */
+  static fromJSON(config) {
+    const { name, description } = config;
+    const vc = new VitessceConfig(name, description);
+    config.datasets.forEach((d) => {
+      const newDataset = vc.addDataset(d.name, d.uid);
+      d.files.forEach((f) => {
+        newDataset.addFile(
+          f.url,
+          f.type,
+          f.fileType,
+        );
+      });
+    });
```

Comment:
Relatedly, I think this might fail?  `f.url` doesn't exist but I think `f.file.url` does?

Reply:
I don't think this is covered in the tests either.

Reply:
Here `f` is just a JSON object from a JSON-based view config (not an instance of `VitessceConfigDatasetFile`)

Reply:
Yup, my bad.